native/core/src/activities.rs

```diff
@@ -123,9 +132,34 @@ impl Activity {
 
     pub fn type_str(&self) -> String {
         match &self.inner {
-            ActivityContent::MembershipChange(c) => c.as_str(),
-            ActivityContent::RoomCreate(_) => "roomCreate",
-            ActivityContent::RoomName(_) => "roomName",
+            ActivityContent::ProfileChange { .. } => "profileChange",
+            ActivityContent::MembershipChange { .. } => "membershipChange",
+
+            ActivityContent::RoomState(OtherState::PolicyRuleRoom(_)) => "policyRuleRoom",
+            ActivityContent::RoomState(OtherState::PolicyRuleServer(_)) => "policyRuleServer",
+            ActivityContent::RoomState(OtherState::PolicyRuleUser(_)) => "policyRuleUser",
+            ActivityContent::RoomState(OtherState::RoomAliases(_)) => "roomAliases",
+            ActivityContent::RoomState(OtherState::RoomAvatar(_)) => "roomAvatar",
+            ActivityContent::RoomState(OtherState::RoomCanonicalAlias(_)) => "roomCanonicalAlias",
+            ActivityContent::RoomState(OtherState::RoomCreate(_)) => "roomCreate",
+            ActivityContent::RoomState(OtherState::RoomEncryption(_)) => "roomEncryption",
+            ActivityContent::RoomState(OtherState::RoomGuestAccess(_)) => "roomGuestAccess",
+            ActivityContent::RoomState(OtherState::RoomHistoryVisibility(_)) => {
+                "roomHistoryVisibility"
+            }
+            ActivityContent::RoomState(OtherState::RoomJoinRules(_)) => "roomJoinRules",
+            ActivityContent::RoomState(OtherState::RoomName(_)) => "roomName",
+            ActivityContent::RoomState(OtherState::RoomPinnedEvents(_)) => "roomPinnedEvents",
+            ActivityContent::RoomState(OtherState::RoomPowerLevels(_)) => "roomPowerLevels",
+            ActivityContent::RoomState(OtherState::RoomServerAcl(_)) => "roomServerAcl",
+            ActivityContent::RoomState(OtherState::RoomThirdPartyInvite(_)) => {
+                "roomThirdPartyInvite"
+            }
+            ActivityContent::RoomState(OtherState::RoomTombstone(_)) => "roomTombstone",
+            ActivityContent::RoomState(OtherState::RoomTopic(_)) => "roomTopic",
+            ActivityContent::RoomState(OtherState::SpaceChild(_)) => "spaceChild",
+            ActivityContent::RoomState(OtherState::SpaceParent(_)) => "spaceParent",
+
```

Comment:
as said above, we decided for a highly-specific-naming approach while most of these aren't that specific but rather descriptive. Has a roomName been set or unset? has the topic been changed or removed?